apis/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -102,50 +119,56 @@ type ResourceLink struct {
 	// TunnelEndpoint link.
-	TunnelEndpoint ResourceLink `json:"tunnelEndpoint"`
+	TunnelEndpoint ResourceLink `json:"tunnelEndpoint,omitempty"`
 }
 
 // Outgoing contains the status of the outgoing peering.
 type Outgoing struct {
 	// Indicates if peering request has been created and this remote cluster is sharing its resources to us.
-	Joined bool `json:"joined"`
-	// Name of created PR.
+	// +kubebuilder:validation:Enum="None";"Pending";"Established";"Deleting"
+	// +kubebuilder:default="None"
+	JoinPhase JoinPhase `json:"joinPhase,omitempty"`
+	// Name of created PR. (Deprecated)
 	RemotePeeringRequestName string `json:"remote-peering-request-name,omitempty"`
-	// Object Reference to created Advertisement CR.
+	// Object Reference to created Advertisement CR. (Deprecated)
 	Advertisement *v1.ObjectReference `json:"advertisement,omitempty"`
-	// Indicates if related identity is available.
+	// Indicates if related identity is available. (Deprecated)
 	AvailableIdentity bool `json:"availableIdentity,omitempty"`
-	// Object reference to related identity.
+	// Object reference to related identity. (Deprecated)
 	IdentityRef *v1.ObjectReference `json:"identityRef,omitempty"`
-	// Advertisement status.
+	// Advertisement status. (Deprecated)
 	AdvertisementStatus advtypes.AdvPhase `json:"advertisementStatus,omitempty"`
 }
 
 // Incoming contains the status of the incoming peering.
 type Incoming struct {
 	// Indicates if peering request has been created and this remote cluster is using our local resources.
-	Joined bool `json:"joined"`
-	// Object Reference to created PeeringRequest CR.
+	// +kubebuilder:validation:Enum="None";"Pending";"Established";"Deleting"
+	// +kubebuilder:default="None"
+	JoinPhase JoinPhase `json:"joinPhase,omitempty"`
+	// Object Reference to created PeeringRequest CR. (Deprecated)
 	PeeringRequest *v1.ObjectReference `json:"peeringRequest,omitempty"`
-	// Indicates if related identity is available.
+	// Indicates if related identity is available. (Deprecated)
 	AvailableIdentity bool `json:"availableIdentity,omitempty"`
-	// Object reference to related identity.
+	// Object reference to related identity. (Deprecated)
 	IdentityRef *v1.ObjectReference `json:"identityRef,omitempty"`
-	// Status of Advertisement created from this PeeringRequest.
+	// Status of Advertisement created from this PeeringRequest. (Deprecated)
 	AdvertisementStatus advtypes.AdvPhase `json:"advertisementStatus,omitempty"`
 }
 
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
 
 // ForeignCluster is the Schema for the foreignclusters API.
-// +kubebuilder:printcolumn:name="Outgoing joined",type=string,JSONPath=`.status.outgoing.joined`
-// +kubebuilder:printcolumn:name="Incoming joined",type=string,JSONPath=`.status.incoming.joined`
+// +kubebuilder:printcolumn:name="Outgoing joined",type=string,JSONPath=`.status.outgoing.joinPhase`
```

Comment:
```suggestion
// +kubebuilder:printcolumn:name="Outgoing peering phase",type=string,JSONPath=`.status.outgoing.joinPhase`
```

Seems more clear to me. Same for the one below.